Glove/fingerprint stuff.

chaoko99 opened the pull request:
🆑 
rscadd: Fingerprints will no longer magically phase through gloves.
tweak: Robolimbs don't make fingerprints anymore.
bugfix: Thick gloves actually use thickmaterial.
/ 🆑 
[why] Explain to me why a fingerprint could go through 2mm of fabric but not a super thin film of latex?

- [x] Produce testing video. https://streamable.com/pyfhb

PsiOmegaDelta commented:
Heads up reviewers, we've got another IPC boost.

PsiOmegaDelta commented on code/modules/detectivework/forensics.dm:
```diff
@@ -58,7 +58,7 @@ atom/var/var/fingerprintslast = null
 		if (H.gloves && H.gloves.body_parts_covered & HANDS && H.gloves != src)
 			H.gloves.add_fingerprint(M)
-			if(!istype(H.gloves, /obj/item/clothing/gloves/latex))
+			if(!istype(H.gloves, /obj/item/clothing/gloves)) //Latex gloves used to be the only effective method of stopping fingerprints. This was stupid. Also lets fingerprints go through rings. NOTE: CHANGE THIS IF YOU'RE READDING FINGERLESS GLOVES.
```
Clipped gloves do exist, yes.

CrimsonShrike replied:
Alright I think I'm going insane, but isn't this check making it so that if you are wearing any type of glove this check fails and thus your prints get added regardless of gloves?

chaoko99 committed:
Fingerprints will no longer magically phase through gloves.
Robolimbs don't make fingerprints anymore.
Thick gloves actually use thickmaterial.

Casper3667 commented:
So if I understand this correctly, any "modified" gloves that xenos always use will now leave fingerprints? Purely curious if I got it right..

chaoko99 commented:
They should.

Zero-Bits commented:
I like this.

chaoko99 commented:
Confirmed working. Note: It will still fingerprint things that are on your person when they spawn because of course it does.

chaoko99 commented:
Tested and confirmed working.

PsiOmegaDelta commented on code/modules/detectivework/forensics.dm:
```diff
@@ -57,11 +57,13 @@ atom/var/var/fingerprintslast = null
-			H.gloves.add_fingerprint(M)
-			if(!istype(H.gloves, /obj/item/clothing/gloves/latex))
-				return 0
+			if(istype(H.gloves, /obj/item/clothing/gloves)) //Latex gloves used to be the only effective method of stopping fingerprints. This was stupid. Also lets fingerprints go through rings. NOTE: CHANGE THIS IF YOU'RE READDING FINGERLESS GLOVES.
```
The only potentially meaningful part of this comment is the `NOTE: [...]` section.

CakeQ commented:
> tweak: Robolimbs don't make fingerprints anymore.

YES

chaoko99 committed:
Update forensics.dm
